### PR TITLE
Fix some edit.js bugs

### PR DESCRIPTION
**Pull request by imgntn**
This PR fixes a few bugs and handles more edge cases related to changing userdata.

To test: 

1. Create a cube
2. Toggle its grabbable, triggerable, and ignore inverse kinematics checkboxes and note that the json updates to match.  note: if unchecked, the ignoreIK key will be removed, not set to false.
3. add a new entry to the json at the same level as the grabbable key.  toggle the boxes from step 2 and ensure this key is not written over.
4. normal object property checkboxes like visible, locked, dynamic and collisonless should work as expected

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9be073cc12c014c5c636f18c4d18c758eb9268ae available at: https://gist.github.com/eabb475174de0d0ef00206916dcb4032<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8529-9be073c.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8529-9be073c.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: ee190db5eb8fa4bbbd416b9a8fe6fde32a633e37 available at: https://gist.github.com/eabb475174de0d0ef00206916dcb4032<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8529-ee190db.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8529-ee190db.dmg

File: scripts/system/edit.js [hunk 1]
```diff
@@ -1369,7 +1369,12 @@ var PropertiesTool = function (opts) {
     });
 
     webView.webEventReceived.connect(function (data) {
-        data = JSON.parse(data);
+        try{
```
Comment (sethalves) [inline scripts/system/edit.js]: space before {

File: scripts/system/edit.js [hunk 2]
```diff
@@ -1369,7 +1369,12 @@ var PropertiesTool = function (opts) {
...
+            data = JSON.parse(data);
+        }
+        catch(e){
```
Comment (sethalves) [inline scripts/system/edit.js]: here too

File: scripts/system/edit.js [hunk 3]
```diff
@@ -1418,7 +1423,9 @@ var PropertiesTool = function (opts) {
             pushCommandForSelections();
             selectionManager._update();
         } else if(data.type === 'saveUserData'){
-            Entities.editEntity(data.id, data.properties)
+            //the event bridge and json parsing handle our avatar id string differently.
+            var actualID = data.id.split('"')[1]
+            var success = Entities.editEntity(actualID, data.properties)
```
Comment (sethalves) [inline scripts/system/edit.js]: is success used anywhere?

Comment (imgntn) [inline scripts/system/edit.js]: i used to print it afterward to tell if it worked but i removed that.  sooo i'll cleanup,  thanks!

File: scripts/system/edit.js [hunk 4]
```diff
@@ -1369,7 +1369,12 @@ var PropertiesTool = function (opts) {
     });
 
     webView.webEventReceived.connect(function (data) {
-        data = JSON.parse(data);
+        try {
+            data = JSON.parse(data);
+        }
+        catch(e) {
+            return;
```
Comment (sethalves) [inline scripts/system/edit.js]: would it make sense to print something here?  silent failures can be hard to track down

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4c63ad1ac866d0d056e124efd7bb7a11776e3061 available at: https://gist.github.com/eabb475174de0d0ef00206916dcb4032<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8529-4c63ad1.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8529-4c63ad1.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: f6bd04780d95f7906581e3bc794dccbf26c7f554 available at: https://gist.github.com/eabb475174de0d0ef00206916dcb4032<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8529-f6bd047.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8529-f6bd047.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 53a22819180a37271c5d1bac25fae8c000268fb4 available at: https://gist.github.com/eabb475174de0d0ef00206916dcb4032<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8529-53a2281.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8529-53a2281.dmg

**Comment by hevyw8** [comment]
# QA Test

## Comments:
Not sure if this is expected behavior but when locking the property as directed in step 4 the json editor reverts to raw json.

## Criteria:

1. Create a cube **_PASS_**
2. Toggle its grabbable, triggerable, and ignore inverse kinematics checkboxes and note that the json updates to match. note: if unchecked, the ignoreIK key will be removed, not set to false. **_PASS_**
3. add a new entry to the json at the same level as the grabbable key. toggle the boxes from step 2 and ensure this key is not written over. **_PASS_**
4. normal object property checkboxes like visible, locked, dynamic and collisonless should work as expected **_SEE COMMENTS_**

### Test system:

system: win10-64
gpu: nvidia gtx1060
cpu: i5-6500 @ 3.2

**Comment by imgntn** [comment]
@hevyw8 -- yeah -- when you lock it, it shows you an uneditable representation of the json as a string.  its basically an overlay over the editor blocking you from clicking on it.
